### PR TITLE
fix: added method and roles check together

### PR DESCRIPTION
**Pull request by AntonioRodriguezRuiz**
This pull request includes updates to the `authorizeRequest` function in the `authMiddleware.js` file and corresponding unit tests in the `authMiddleware.test.js` file. The main changes involve adding method-specific permission checks and updating test cases to reflect these changes.

### Updates to Authorization Logic:

* [`src/middleware/authMiddleware.js`](diffhunk://#diff-b03519de8bc8a3860fadfeaedb79a86b62610df6361695aa44da0577523a3ea8L71-R91): Modified the `authorizeRequest` function to include method-specific permission checks, ensuring that permissions are validated based on the HTTP method of the request.

### Updates to Unit Tests:

* [`tests/unit/middleware/authMiddleware.test.js`](diffhunk://#diff-432bda35977bc443f15aab0b9af6906d7aed1f4de529aa91e6cb7337dfb03144L182-R182): Updated multiple test cases to use a consistent user ID (`clinicadminId`) for the mocked `User.findById` method. [[1]](diffhunk://#diff-432bda35977bc443f15aab0b9af6906d7aed1f4de529aa91e6cb7337dfb03144L182-R182) [[2]](diffhunk://#diff-432bda35977bc443f15aab0b9af6906d7aed1f4de529aa91e6cb7337dfb03144L211-R211) [[3]](diffhunk://#diff-432bda35977bc443f15aab0b9af6906d7aed1f4de529aa91e6cb7337dfb03144L242-R242)
* [`tests/unit/middleware/authMiddleware.test.js`](diffhunk://#diff-432bda35977bc443f15aab0b9af6906d7aed1f4de529aa91e6cb7337dfb03144R258-R286): Added a new test case to verify that a 403 status is returned if a user has permissions on a target user for one method but not for another.

### CODE REVIEW

File: src/middleware/authMiddleware.js [hunk 1]
```diff
@@ -68,23 +68,28 @@ export const authorizeRequest = (method) => {
...
                         rolesData.some((role) => {
                           return role.permissions.some(
                             (permission) =>
-                              targetUserRoles.every((role) =>
-                                permission.onRoles.includes(role)
-                              )
+                              permission.method === method &&
+                                    targetUserRoles.every((onRole) =>
+                                      permission.onRoles.includes(onRole)
+                                    )
```
Comment (alvgonfri) [inline src/middleware/authMiddleware.js:80]: With this change, I think the following code is no longer necessary:

image

File: src/middleware/authMiddleware.js [hunk 2]
```diff
@@ -68,23 +68,28 @@ export const authorizeRequest = (method) => {
...
                           );
                         });
         } else {
+          console.log('Hi', hasPermission, req.userId, targetUserId);
+          console.log(targetUserRoles);
+          console.log(rolesData);
```
Comment (alvgonfri) [inline src/middleware/authMiddleware.js:86]: Remove this.

**Review by alvgonfri — CHANGES_REQUESTED** [review]

File: src/middleware/authMiddleware.js (hunk 1 above)
Comment (AntonioRodriguezRuiz) [reply src/middleware/authMiddleware.js:80]: I tested it and it is in cases where the endpoint does not include a userId as a parameter

File: src/middleware/authMiddleware.js (hunk 2 above)
Comment (AntonioRodriguezRuiz) [reply src/middleware/authMiddleware.js:86]: I will

**Review by alvgonfri — APPROVED** [review]
Everything correct now.

**Pull request merged**